### ISSUE TITLE
Find an opinionated code formatting tool, like prettier in the JS world

### ISSUE THREAD

**Issue by acelaya** [hypothesis/wp-hypothesis#50]
Prettier has an official plugin to format PHP code which might be a good option: https://github.com/prettier/plugin-php

It has some considerations:

* We need to check if it does not conflict with Wordpress coding standard, and if it does, we may want to move to [PER-2](https://www.php-fig.org/per/coding-style/)/[PSR-12](https://www.php-fig.org/psr/psr-12/) instead 
* Prettier is a JS tool. Since the project already has a package.json, we could incorporate yarn to the toolbelt and workflow without a lot of effort, but may be easier to introduce a PHP-based tool.

**Comment by acelaya** [comment]
> @acelaya don't feel that we have to use Yarn in every project. For small or less-critical packages, npm may be just fine and can keep the toolchain a little simpler. npm's main downside is that it's lockfile format is less diff-friendly and its terminal output is less well presented.

True! But I with "yarn" I actually meant anything "non-composer" 😅 

I'll rewrite it for clarity.

**Comment by SteelWagstaff** [comment]
A lot of folks in the WP environment use PHPCS with the WordPress coding standards: https://github.com/WordPress/WordPress-Coding-Standards. Here's the documentation on official WP coding standards, which are encouraged for plugins/themes as well: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/

**Comment by acelaya** [comment]
> A lot of folks in the WP environment use (PHP CodeSniffer)[https://github.com/squizlabs/PHP_CodeSniffer] with the WordPress coding standards: https://github.com/WordPress/WordPress-Coding-Standards. Here's the documentation on official WP coding standards, which are encouraged for plugins/themes as well: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/

That's what the project is currently using, but we need to make sure it fits nicely with the other hypothesis projects, where Python and JS/TS is primarily used.

It's important that anyone on the team can jump-in to work on this and the tooling and workflow feels familiar.

That's preferable even if it deviates from standard WordPress community "ways".
